src/test/java/uk/nhs/hee/tis/revalidation/core/service/AdminServiceTest.java

```diff
@@ -80,43 +84,56 @@ void shouldReturnEmptyListWhenNoAssignableAdminsFound() {
-
-    AttributeType attributeType4 = new AttributeType();
-    attributeType4.setName("given_name");
-    attributeType4.setValue("user2");
-
-    AttributeType attributeType5 = new AttributeType();
-    attributeType5.setName("email");
-    attributeType5.setValue("user1@email.com");
-
-    AttributeType attributeType6 = new AttributeType();
-    attributeType6.setName("email");
-    attributeType6.setValue("user2@email.com");
-
-    user1.setAttributes(Arrays.asList(attributeType1, attributeType3, attributeType5));
-    user2.setAttributes(Arrays.asList(attributeType2, attributeType4, attributeType6));
-
-    ListUsersInGroupResult listUsersInGroupResult = new ListUsersInGroupResult();
-    listUsersInGroupResult.setUsers(Arrays.asList(user1, user2));
-
-    when(identityProvider.listUsersInGroup(any())).thenReturn(listUsersInGroupResult);
+    UserType user1 = UserType.builder()
+        .build();
+    user1 = user1.toBuilder().username("user1").build();
+
+    UserType user2 = UserType.builder()
+        .build();
+    user2 = user2.toBuilder().username("user2").build();
+
+    AttributeType attributeType1 = AttributeType.builder()
+        .build();
+    attributeType1 = attributeType1.toBuilder().name("family_name").build();
+    attributeType1 = attributeType1.toBuilder().value("user1").build();
+
+    AttributeType attributeType2 = AttributeType.builder()
+        .build();
+    attributeType2 = attributeType2.toBuilder().name("family_name").build();
+    attributeType2 = attributeType2.toBuilder().value("user2").build();
+
+    AttributeType attributeType3 = AttributeType.builder()
+        .build();
+    attributeType3 = attributeType3.toBuilder().name("given_name").build();
+    attributeType3 = attributeType3.toBuilder().value("user1").build();
+
+    AttributeType attributeType4 = AttributeType.builder()
+        .build();
+    attributeType4 = attributeType4.toBuilder().name("given_name").build();
+    attributeType4 = attributeType4.toBuilder().value("user2").build();
+
+    AttributeType attributeType5 = AttributeType.builder()
+        .build();
+    attributeType5 = attributeType5.toBuilder().name("email").build();
+    attributeType5 = attributeType5.toBuilder().value("user1@email.com").build();
+
+    AttributeType attributeType6 = AttributeType.builder()
+        .build();
+    attributeType6 = attributeType6.toBuilder().name("email").build();
+    attributeType6 = attributeType6.toBuilder().value("user2@email.com").build();
+
+    user1 = user1.toBuilder()
+        .attributes(Arrays.asList(attributeType1, attributeType3, attributeType5)).build();
+    user2 = user2.toBuilder()
+        .attributes(Arrays.asList(attributeType2, attributeType4, attributeType6)).build();
+
+    ListUsersInGroupResponse listUsersInGroupResult = ListUsersInGroupResponse.builder()
+        .build();
+    listUsersInGroupResult = listUsersInGroupResult.toBuilder().users(Arrays.asList(user1, user2))
+        .build();
```

Comment:
Again, condense these to use the builder pattern properly.

Reply:
Looks like this is still outstanding.  I'd recommend reading it through and identifying it but spotted what could be a check if you are really struggling with the builder pattern.

Reply:
I've noted what's still outstanding